app/src/main/java/org/schabi/newpipe/local/history/StatisticsPlaylistFragment.java

```diff
@@ -366,6 +367,19 @@ private void showStreamDialog(final StreamStatisticsEntry item) {
         if (KoreUtils.shouldShowPlayWithKodi(context, infoItem.getServiceId())) {
             entries.add(StreamDialogEntry.play_with_kodi);
         }
+
+        // show "mark as watched" only when watch history is enabled
+        final boolean isWatchHistoryEnabled = PreferenceManager
+                .getDefaultSharedPreferences(context)
+                .getBoolean(getString(R.string.enable_watch_history_key), false);
+        if (item.getStreamEntity().getStreamType() != StreamType.AUDIO_LIVE_STREAM
+                && item.getStreamEntity().getStreamType() != StreamType.LIVE_STREAM
+                && isWatchHistoryEnabled
+        ) {
```

Comment:
This code should definitely be deduplicated. I'd create a static method `shouldAddMarkAsWatched(StreamType)` to `StreamDialogEntry` that replaces all of these lines and is also used in `FeedFragment`.

Reply:
Fair enough, that's done. I had to pass the context in as well to get `isWatchHistoryEnabled` working.